src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -178,4 +178,32 @@ export class SearchLogGroupWizard extends Wizard<SearchLogGroupWizardResponse> {
         this.form.filterPattern.bindPrompter(createFilterpatternPrompter)
         this.form.timeRange.bindPrompter(() => new TimeFilterSubmenu())
     }
+}
+
+export class JumpToStream implements vscode.DefinitionProvider {
+    private registry: LogStreamRegistry
+
+    public constructor(registry: LogStreamRegistry) {
+        this.registry = registry
+    }
+    provideDefinition(
```

Comment:
This can be `async`:
```suggestion
    async provideDefinition(
```

You'll need to change the return type: 
```ts
Promise<vscode.Definition | vscode.LocationLink[] | undefined>
```

Reply:
This is what I originally tried and it gives me a linting error. It still runs and actually gives me the exact behavior I want however. It says `Property 'provideDefinition' in type 'JumpToStream' is not assignable to the same property in base type 'DefinitionProvider'.`

Reply:
> It still runs and actually gives me the exact behavior I want however.

This is expected. **Typescript types do not affect runtime behavior**

> Why do I remove the providerResult layer when it becomes a promise?

Because `ProviderResult<T>` is equivalent to:
```ts
T | undefined | null | Thenable<T | undefined | null>
```

We don't need to care about `Thenable` or `null` so we throw it away. The VS Code API is just saying it can support all these different types, but we only need to return a subset of these types.